File: easy/3345-smallest-divisible-digit-product-i/solution.java
/**
 * 3345. Smallest Divisible Digit Product I
 * Time: O(A * d)
 * Space: O(1)
 */
class Solution {
    public int smallestNumber(int n, int t) {
        int x = n;
        while (true) {
            int product = 1;
            for (int y = x; y > 0; y /= 10) {
                product *= y % 10;
            }
            if (product % t == 0) {
                return x;
            }
            x++;
        }
    }
}

File: easy/3345-smallest-divisible-digit-product-i/solution.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        """
        3345. Smallest Divisible Digit Product I
        Time: O(A * d)
        Space: O(1)
        """
        x = n
        while True:
            product = 1
            for ch in str(x):
                product *= int(ch)
            if product % t == 0:
                return x
            x += 1
